org.jcryptool.core/src/org/jcryptool/core/Application.java

```diff
@@ -9,75 +9,385 @@
+		List<String> commandArgs = getCommandLinePartFromString(System.getProperty("eclipse.commands"));
+		List<String> reconstructedCmdline = new LinkedList<>();
+		reconstructedCmdline.addAll(vmArg);
+		reconstructedCmdline.addAll(vmArgsArg);
+		reconstructedCmdline.addAll(commandArgs);
+		List<String> filteredCommandline = filter.apply(reconstructedCmdline); // filtering is necessary _exactly_ here, not after next two lines. TODO maybe: leave out vmargs out of filtering process (less power, but less errorprone also)
+		filteredCommandline.add("-vmargs"); // this is somehow necessary to tell the next instance what the VM arguments were...
+		filteredCommandline.addAll(vmArgsArg); // see above line
+		String resultargs = filteredCommandline.stream().collect(Collectors.joining("\n"));
+		System.err.println(String.format(
+				"Restarting the application with filtered commandline, now stored in eclipse.exitdata: %s",
+				resultargs));
+		System.setProperty("eclipse.exitdata", resultargs);
+	}
+
+	/**
+	 * This restarts the application. May only be called when the workbench has been
+	 * started yet! Else, see the code pertaining to
+	 * {@link Application#restart_cmdlinefilter} and
+	 * {@link #startApplication(IApplicationContext)}
+	 * 
+	 * @param ask           whether a dialog box should be opened that asks whether
+	 *                      the action should be performed.
+	 * @param cmdlineFilter a filter that is applied to the commandline arguments
+	 */
+	private static void restartAppWithCommandlinefilter(boolean ask,
+			Function<List<String>, List<String>> cmdlineFilter) {
+		boolean doRestart = false;
+		if (ask) {
+			MessageBox mbox = new MessageBox(Display.getDefault().getActiveShell(),
+					SWT.ICON_QUESTION | SWT.YES | SWT.NO);
+			mbox.setText(org.jcryptool.core.preferences.pages.Messages.MessageTitleRestart);
+			mbox.setMessage(org.jcryptool.core.preferences.pages.Messages.MessageRestart);
+			if (mbox.open() == SWT.YES) {
+				LogUtil.logInfo(org.jcryptool.core.preferences.pages.Messages.MessageLogRestart);
+				doRestart = true;
+			}
+		} else {
+			doRestart = true;
+		}
+
+		if (doRestart) {
+			// this instructs the application to filter commandline arguments and
+			// relaunching instead of restarting
+			Application.restart_cmdlinefilter = cmdlineFilter;
+			PlatformUI.getWorkbench().restart();
+		}
+	}
+
+	/**
+	 * This takes a filter that turns one commandline, or .ini-file content, into
+	 * another. Applies it to the JCrypTool inifile. Backs the inifile up before.
+	 * 
+	 * Older inifile backups are deleted without asking.
+	 * 
+	 * @throws IOException when something goes wrong with the .ini-file rewrite or
+	 *                     the backup process
+	 */
+	private static void applyFilterToInifile(Function<List<String>, List<String>> filter) throws IOException {
+		String path = Platform.getInstallLocation().getURL().toExternalForm();
+		String fileNameOrg = Platform.getProduct().getName() + ".ini"; //$NON-NLS-1$
```

Comment:
I personally dislike the `//$NON-NLS-N$` comments, as they add nothing to the code but unnecessary clutter. Maybe this could be point of a further coding style / coding convention discussion.

Reply:
Completely agree, must be from some ... > Source > Format. Is the checkstyle PR affecting this behavior perhaps?